cluster-agent/controllers/argoproj.io/application_controller.go

```diff
@@ -163,6 +169,153 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
+					strconv.Itoa(offSet)+" to: "+strconv.Itoa(offSet+batchSize))
+				break
+			}
+
+			// Break the loop if no entries are left in table to be processed.
+			if len(listOfApplicationsFromDB) == 0 {
+				log.Info("All Application entries are processed by Namespace Reconciler.")
+				break
+			}
+
+			// Iterate over batch received above.
+			for _, applicationRowFromDB := range listOfApplicationsFromDB {
+
+				// Fetch the Application object from DB
+				if err := yaml.Unmarshal([]byte(applicationRowFromDB.Spec_field), &applicationFromDB); err != nil {
+					log.Error(err, "Error occurred in Namespace Reconciler while unmarshalling application: "+applicationRowFromDB.Application_id)
+					continue // Skip to next iteration instead of stopping the entire loop.
+				}
+
+				// Fetch the Application object from k8s
+				applicationFromArgoCD := appv1.Application{}
+				namespacedName := types.NamespacedName{
+					Name:      applicationFromDB.Name,
+					Namespace: applicationFromDB.Namespace}
+
+				err := r.Get(ctx, namespacedName, &applicationFromArgoCD)
+				if err != nil {
+					if strings.Contains(err.Error(), "not found") {
+						log.Info("Application " + applicationRowFromDB.Application_id + " not found in ArgoCD, probably user deleted it, " +
+							"but It still exists in DB, hence recreating application in ArgoCD.")
+
+						// Assign values to ArgoCD application object from DB object
+						applicationFromArgoCD.Kind, applicationFromArgoCD.APIVersion,
+							applicationFromArgoCD.Name, applicationFromArgoCD.Namespace,
+							applicationFromArgoCD.Spec.Source.RepoURL, applicationFromArgoCD.Spec.Source.Path,
+							applicationFromArgoCD.Spec.Source.TargetRevision = applicationFromDB.Kind,
+							applicationFromDB.APIVersion, applicationFromDB.Name,
+							applicationFromDB.Namespace, applicationFromDB.Spec.Source.RepoURL,
+							applicationFromDB.Spec.Source.Path, applicationFromDB.Spec.Source.TargetRevision
+
+						if applicationFromDB.Spec.SyncPolicy != nil {
+							applicationFromArgoCD.Spec.SyncPolicy = &appv1.SyncPolicy{
+								Automated: &appv1.SyncPolicyAutomated{Prune: applicationFromDB.Spec.SyncPolicy.Automated.Prune},
+							}
+						}
+
+						// Recreate ArgoCD Application
+						if err := r.Create(ctx, &applicationFromArgoCD); err != nil {
+							log.Error(err, "Namespace Reconciler failed to create application in ArgoCD.")
+							continue
+						}
+					} else {
+						log.Error(err, "Error occurred in Namespace Reconciler while fetching application from cluster: "+applicationRowFromDB.Application_id)
+						continue
+					}
+				}
+
+				// At this point we have the applications from ArgoCD and DB, now compare them to check if they are not in Sync.
+				if CompareApplications(applicationFromArgoCD, applicationFromDB, log) {
+					log.Info("Argo application is in Sync with DB, Application:" + applicationRowFromDB.Application_id)
```

Comment:
Can change this to log at debug level, because I suspect this will be very verbose with a large number of Applications.